modules/videoio/src/cap_v4l.cpp

```diff
@@ -1546,6 +1557,11 @@ static IplImage* icvRetrieveFrameCAM_V4L( CvCaptureCAM_V4L* capture, int) {
                 (unsigned char*)capture->buffers[(capture->bufferIndex+1) % capture->req.count].start,
                 (unsigned char*)capture->frame.imageData);
         break;
+    case V4L2_PIX_FMT_Y16:
+        memcpy((char *)capture->frame.imageData,
+               (char *)capture->buffers[capture->bufferIndex].start,
+               capture->frame.imageSize);
+        break;
```

Comment:
Here you can add branch for "default" BGR mode under `if (capture->convert_rgb)`